src/net.md

```diff
@@ -1123,6 +1124,102 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-handle-rate-limited-api]: #ex-handle-rate-limited-api
+<a name="ex-handle-rate-limited-api"/>
+## Handle a rate-limited API
+
+[![reqwest-badge]][reqwest] [![hyper-badge]][hyper] [![cat-net-badge]][cat-net]
+
+This example uses the [GitHub API - Rate limiting], as an example of how to handle error conditions.
+
+For that defines three custom headers using the [`hyper::header!`] macro to read the values sent by
+the server, and checks for [`reqwest::StatusCode::Forbidden`]. If the rate limit was excided raises
+an exception with a detailed message containing the information provided in the custom headers.
+
+```rust,no_run
+#[macro_use]
+extern crate error_chain;
+#[macro_use]
+extern crate hyper;
+extern crate reqwest;
+# extern crate rayon;
+
+use std::time::{Duration, UNIX_EPOCH};
+
+use reqwest::StatusCode;
+# use rayon::prelude::*;
+
+error_chain! {
+    errors {
+        RateLimitExceded(quote: usize, lease_secs: u64) {
+            description("rate limit exceeded")
+            display("rate limit exceeded the {} calls, the next call will be allowed within {} seconds", quote, lease_secs)
+        }
+    }
+#    foreign_links {
+#        Io(std::io::Error);
+#        Time(std::time::SystemTimeError);
+#        Reqwest(reqwest::Error);
+#    }
+}
+
+header! { (XRateLimitLimit, "X-RateLimit-Limit") => [usize] }
+header! { (XRateLimitRemaining, "X-RateLimit-Remaining") => [usize] }
+header! { (XRateLimitReset, "X-RateLimit-Reset") => [u64] }
+
+fn run() -> Result<()> {
+    let url = "https://api.github.com/users/rust-lang-nursery ";
+    let client = reqwest::Client::new();
+
+#   let response = client.get(url).send()?;
+#
+#   // for the purpose of this example will exaust the limit off calls
+#   if response.status() == StatusCode::Ok {
+#       let rate_remaining = response.headers().get::<XRateLimitRemaining>().ok_or(
+#           "response doesn't include the expected X-RateLimit-Remaining header",
+#       )?;
+#
+#       (0..**rate_remaining).into_par_iter().for_each(|_| {
+#           client.get(url).send().expect("request failed");
+#       });
+#   }
+#
+    let response = client.get(url).send()?;
+
+    let rate_limit = response.headers().get::<XRateLimitLimit>().ok_or(
+        "response doesn't include the expected X-RateLimit-Limit header",
+    )?;
+
+    let rate_remaining = response.headers().get::<XRateLimitRemaining>().ok_or(
+        "response doesn't include the expected X-RateLimit-Remaining header",
+    )?;
+
+    let rate_reset_at = response.headers().get::<XRateLimitReset>().ok_or(
+        "response doesn't include the expected X-RateLimit-Reset header",
+    )?;
+
+    let rate_reset_within = Duration::from_secs(**rate_reset_at) - UNIX_EPOCH.elapsed()?;
+
+    if response.status() == StatusCode::Forbidden && **rate_remaining == 0 {
+        bail!(ErrorKind::RateLimitExceded(
+            **rate_limit,
+            rate_reset_within.as_secs(),
+        ));
+    }
+
+    println!(
```

Comment:
I'd suggest we try to handle the ratelimit (possibly wait and retry?) here instead of just printing and bailing.

I know that some other contributors had opinions on this slightly unclear example idea.
@konstin and @clarcharr  would like to add some thoughts on this example?

Reply:
I'll wait a few days to get more clear directions what do you wanna have in the cookbook, thank you.

Reply:
@jbolila looks like there are no suggestions ;) So lets go with the simplest possible implementation (wait and retry).